spread/qemu.go

```diff
@@ -114,10 +116,18 @@ func (p *qemuProvider) Allocate(ctx context.Context, system *System) (Server, er
 	serial := fmt.Sprintf("telnet:127.0.0.1:%d,server,nowait", port+100)
 	monitor := fmt.Sprintf("telnet:127.0.0.1:%d,server,nowait", port+200)
 	fwd := fmt.Sprintf("user,hostfwd=tcp:127.0.0.1:%d-:22", port)
-	cmd := exec.Command("qemu-system-x86_64", "-enable-kvm", "-snapshot", "-m", strconv.Itoa(mem), "-net", "nic", "-net", fwd, "-serial", serial, "-monitor", monitor, path)
+	cmd := exec.Command(qemuBinary, "-enable-kvm", "-snapshot", "-m", strconv.Itoa(mem), "-net", "nic", "-net", fwd, "-serial", serial, "-monitor", monitor, path)
 	if os.Getenv("SPREAD_QEMU_GUI") != "1" {
 		cmd.Args = append([]string{cmd.Args[0], "-nographic"}, cmd.Args[1:]...)
 	}
+	switch system.Bios {
+	case "legacy", "":
+		// nothing to do, that is the qemu default
+	case "uefi":
+		cmd.Args = append([]string{cmd.Args[0], "-bios", "/usr/share/OVMF/OVMF_CODE.fd"}, cmd.Args[1:]...)
```

Comment:
This will be a problem across distributions, as the name is not standardized.

My advice is as follows:

1) Readlink /usr/share/spread/qemu-efi-firmware
2) If present, that's the name
3) If the link is not present, use a hard-coded default.

This way any packaged version of spread can simply drop a symbolic link and depend on the right firmware package.

Reply:
The default should work on ubuntu/debian/fedora but it seems it will not on suse so I guess that advise makes sense.

Reply:
It can be handled with distro-patch as well so that's not all terrible.